File: 4/4.php
<?php
$horas = readline("Introduce numero de horas trabajadas: ");
$costo = readline("Introduce costo por horas: ");

if(is_numeric($horas) && is_numeric($costo)){
    $total = intval($horas) * floatval($costo);
    echo "El total del pago $ ".round($total,2);
}else{
    echo "Los valores deben ser numéricos";
}

File: 4/4.py

try:
    horas = int(input("Introduce numero de horas trabajadas: "))
    costo = float(input("Introduce costo por horas: "))    
    if(type(horas)==int and type(costo)==float):
        total = horas * costo
        print("El total del pago $ "+ str(round(total,2)))
except ValueError:
    print("Los valores deben ser numéricos")    